hack/generator/pkg/astmodel/optional_type.go

```diff
@@ -19,16 +19,32 @@ type OptionalType struct {
 	element Type
 }
 
-var _ Type = &OptionalType{}
+// type assertions
+var (
+	_ Type     = &OptionalType{}
+	_ MetaType = &OptionalType{}
+)
 
-var _ MetaType = &OptionalType{}
+// cache of commonly-used values
+var cachedOptionals = map[Type]*OptionalType{
+	BoolType:   {BoolType},
+	FloatType:  {FloatType},
+	IntType:    {IntType},
+	StringType: {StringType},
+	UInt32Type: {UInt32Type},
+	UInt64Type: {UInt64Type},
+}
```

Comment:
I just thought it might be nice to help avoid many “duplicate” instantiations, and will speed up equality checks and suchlike slightly. From my numbers it saves about five thousand allocations.